backend/httpclient/http_client.go

```diff
@@ -84,6 +86,13 @@ func GetTransport(opts ...Options) (http.RoundTripper, error) {
 		clientOpts.Middlewares = clientOpts.ConfigureMiddleware(clientOpts, clientOpts.Middlewares)
 	}
 
+	if proxy.SecureSocksProxyEnabled(clientOpts.ProxyOptions) {
+		err = proxy.NewSecureSocksHTTPProxy(transport, clientOpts.ProxyOptions)
```

Comment:
Nit. Would it make sense to check if it should be enabled in `proxy.NewSecureSocksHTTPProxy` as well to simplify the API? Then calling this and if not enabled the provided transport is not touched?

Reply:
Good idea! I updated it to be named `ConfigureSecureSocksHTTPProxy` and added the check directly in there